diff -ru libffi-3.0.10/src/arm/sysv.S libffi-3.0.10+iPhone/src/arm/sysv.S
--- libffi-3.0.10/src/arm/sysv.S	2011-08-23 14:27:24.000000000 +0000
+++ libffi-3.0.10+iPhone/src/arm/sysv.S	2012-03-18 23:31:44.000000000 +0000
@@ -187,7 +187,7 @@
 	@     r1 already set
 
 	@ Call ffi_prep_args(stack, &ecif)
-	bl	ffi_prep_args
+	bl	CNAME(ffi_prep_args)
 
 	@ move first 4 parameters in registers
 	ldmia	sp, {r0-r3}
@@ -362,7 +362,7 @@
 	sub	r2, fp, #64   @ VFP scratch space
 
 	@ Call ffi_prep_args(stack, &ecif, vfp_space)
-	bl	ffi_prep_args
+	bl	CNAME(ffi_prep_args)
 
 	@ Load VFP register args if needed
 	cmp	r0, #0
@@ -444,7 +444,7 @@
 	sub	sp, sp, #72
 	str	sp, [sp, #64]
 	add	r1, sp, #64
-	bl	ffi_closure_SYSV_inner
+	bl	CNAME(ffi_closure_SYSV_inner)
 
 	cmp	r0, #FFI_TYPE_INT
 	beq	.Lretint_vfp
